src/storage/FileResourceStore.ts

```diff
@@ -385,16 +373,17 @@ export class FileResourceStore implements ResourceStore {
     } else {
       // If metadata is specified, save it in a corresponding metadata file.
       if (metadata) {
-        await this.createDataFile(joinPath(this.rootFilepath, path, `${resourceName}.metadata`), metadata);
+        await this.createDataFile(this.resourceMapper.getAbsolutePath(path, `${resourceName}.metadata`), metadata);
       }
 
       // If no error thrown from above, indicating failed metadata file creation, create the actual resource file.
       try {
-        await this.createDataFile(joinPath(this.rootFilepath, path, resourceName), data);
-        return { path: this.resourceMapper.mapFilePathToUrl(joinPath(this.rootFilepath, path, resourceName)) };
+        const fullPath = this.resourceMapper.getAbsolutePath(path, resourceName);
+        await this.createDataFile(fullPath, data);
```

Comment:
I only realize this now, but there is no guarantee right now that if a new file gets created with content-type `text/turtle` it is actually going to have the file extension `.ttl` right? (Unless the PUT identifier/POST slug happens to end on it). @RubenVerborgh do we want to add it if necessary?

Reply:
Indeed, and that knowledge should _ONLY_ be in the ResourceMapper. Let's create issue.